### ISSUE TITLE
[Feature Request] Customizable Keyboard Shortcuts

### ISSUE THREAD

**Issue by lyoye2** [cemu-project/Cemu#721]
Hello.
Cemu has a number of Keyboard Shortcuts that are (most of them) hardcoded.

In my humble opinion, I believe that offering the user the ability to remap them to their liking would be a very welcome QoL improvement.

I attach an image in which I offer you my rudimentary vision of how it could be incorporated into the GUI. (I'm aware that it's not very elegant, but I hope it gives you an idea).

In addition, I have removed the 'blow mic' and 'show screen' options from the 'Input Settings' window, and instead moved them along with the rest of the Shortcuts to a tab of their own, which will bring them together in one place, for convenience.

I realize that doing this is a monumental task, so I would like to make it clear that it is not my intention to rush or force anyone to do anything. This is just my suggestion.

Thank you very much for everything.
![Input Settings](https://user-images.githubusercontent.com/129283103/228516788-1c3018c0-d818-4377-8921-f1287080f2d6.png)
![Hotkey Settings](https://user-images.githubusercontent.com/129283103/228516794-157fa988-c72f-4a38-b180-87e990c9cac4.png)

**Comment by rluxv** [comment]
This is something I could try to tackle.

**Comment by SilentException** [comment]
Currently, on Steam Deck, Alt+F4 doesn't close Cemu in game mode. This worked previously but I don't know the exact version it broke.

But perhaps during this revamp of keyboard shortcuts you could also take a look at that. 
Or even better, make "Exit" shortcut configurable as well. Thanks!

**Comment by SilentException** [comment]
@jn64 Yeah I suppose it is SD issue, makes sense because in Desktop mode it works fine.
Would be nice to have the configurable shortcut for Gaming Mode.

**Comment by rode7d** [comment]
Customizable Hotkeys would be indeed a very welcome feature.

**Comment by Lazorne** [comment]
Would love this as well!

**Comment by jn64** [comment]
> Would love this as well!

Please, just add a thumbs-up on the first post, instead of posting this and
sending a notification/email to everyone subscribed to the issue.

**Comment by blakbin** [comment]
But i think it will be useless, if you use Gamepad, because the Second Screen interacting with Mouse

**Comment by muzzol** [comment]
> But i think it will be useless, if you use Gamepad, because the Second Screen interacting with Mouse

I use Steamdeck and it got a touch screen and you can also attach a second monitor so you can play with dual setups. anyway, there's a lot of ways to send input to a program, not just mouse.

allowing custom shortcuts helps a lot with some handhelds and other controllers.

**Comment by Immersion95** [comment]
It would be great to have + having the possibility to bind controller input to thos hotkeys like Dolphin/Pcsx2

Example: Having a combo on the controller allowing to change view would be a godsend

**Comment by zhen-huan-hu** [comment]
Allowing controller combo as hotkeys would be life saving for Steam Deck with a frontend like ES-DE. For example, using start + option combo to gracefully exit the emulator.

**Comment by zhen-huan-hu** [comment]
Wonder if there is any update on this feature request? The hotkey tab disappeared completely in Cemu 2.2 (Windows).

**Comment by persechini** [comment]
For couch gaming this would also be very much appreciated, ES-DE, for instance, doesn't currently have any way of closing emulators using a controller than don't already have their own key bindings for that

**Comment by Hexaae** [comment]
Yes please, add also multiple buttons (6 should be enough) support under HOTKEYS SETTINGS (e.g. press RZ+RB+LZ+LB to quit CEMU)
